### PR TITLE
Removes dependency on finagle and thrift

### PR DESCRIPTION
**Pull request by jcarres-mdsol**
@jfeltesse-mdsol @ykitamura-mdsol @ssteeg-mdsol @adriancole 

I do not know what else is in finagle but it is huge. Removing finagle from a rails project lowered the startup memory from 78MB to 74MB
That's 4 times the memory taken by the mysql adaptor for instance.

Anyways, another good reason is that we were in effect monkey patching a lot of classes. Now we got them complete in this codebase and we can move them out of the terrible high level Trace module in a different PR as well as doing some other cleanups. I did not do that on this PR as it is big enough.

Most code changes is moving finagle code into this codebase.

Specs changes are mostly because we were not spec-ing the right thing. We were testing and modifying Trace.id instead of TraceContainer.current, the thing was working by some miracle.

### CODE REVIEW

**Comment by jfeltesse-mdsol** [comment]
+1000000000 👍

File: lib/zipkin-tracer/trace.rb [hunk 1]
```diff
@@ -1,19 +1,132 @@
...
+        value: @value,
+        timestamp: @timestamp,
+        endpoint: host.to_h
+      }
+    end
+  end
+
+  class BinaryAnnotation
+    SERVER_ADDRESS = 'sa'.freeze
+    URI = 'http.uri'.freeze
```
Comment (ykitamura-mdsol) [inline lib/zipkin-tracer/trace.rb]: how about switching to `http.url`?
https://zipkin.io/public/thrift/v1/zipkinCore.html#Const_HTTP_URL

File: lib/zipkin-tracer/trace.rb [hunk 2]
```diff
@@ -1,19 +1,137 @@
...
+
+  class SpanId
+    HEX_REGEX = /^[a-f0-9]{16,32}$/i
+    MAX_SIGNED_I64 = 9223372036854775807
+    MASK = (2 ** 64) - 1
+
+    def self.from_value(v)
+      if v.is_a?(String) && v =~ HEX_REGEX
+        # drops any bits higher than 64 by selecting right-most 16 characters
+        new(v.length > 16 ? v[v.length - 16, 16].hex : v.hex)
```
[the remark this answers is not in the document]
Comment (ykitamura-mdsol) [reply lib/zipkin-tracer/trace.rb]: it's already done here :p

https://github.com/openzipkin/zipkin-ruby/blob/7d5c91bff56bc312378f90512d08d86730acded8/lib/zipkin-tracer/trace.rb#L161-L173

@jcarres-mdsol can you remove/modify the comment?

File: README.md [hunk 3]
```diff
@@ -124,6 +124,8 @@ responds to #value!, it will be called (to block until completed).
 
 Caveat: Hermann is only usable from within Jruby, due to its implementation of zookeeper based broker discovery being JVM based.
 
+The Kafka transport send data using Thrift. Since version 0.31.0, Thrift is not a dependency, thus the gem 'finagle-thrift' needs to be added to the Gemfile also.
```
[the remark this answers is not in the document]
Comment (jcarres-mdsol) [reply README.md]: we need to move everything to v2, I think this change was blocking it

**Pull request merged**